### PR TITLE
Tag cluster & workspace as beta features in extension api

### PR DESCRIPTION
**Pull request by jakolehm**
Both of those APIs still include internal details that we really have to cleanup. Tagging these as `beta` will set better expectation level and also allows us to tweak these APIs within 4.x series (until we take the beta tag away).

### CODE REVIEW

**Comment by jakolehm** [comment]
Cluster cleanup has been already started but it didn't make it into 4.0: https://github.com/lensapp/lens/pull/1435

File: src/common/workspace-store.ts [hunk 1]
```diff
@@ -26,11 +26,38 @@ export interface WorkspaceState {
...
+/**
+ * Workspace
+ *
+ * @beta
+ */
 export class Workspace implements WorkspaceModel, WorkspaceState {
+  /**
+   * Unique id
+   * @observable
+   */
```
Comment (Nokel81) [inline src/common/workspace-store.ts]: Won't this whole PR conflict with #1573 ?

File: src/common/workspace-store.ts [hunk 2]
```diff
@@ -44,23 +71,39 @@ export class Workspace implements WorkspaceModel, WorkspaceState {
...
   getState(): WorkspaceState {
     return {
       enabled: this.enabled
     };
   }
 
+  /**
+   * Push state
+   *
+   * @interal
```
Comment (Nokel81) [inline src/common/workspace-store.ts:106]: I really think that any API marked as `@internal` should be not exported in a similar style to #1546.

File: src/main/cluster.ts [hunk 3]
```diff
@@ -48,10 +48,31 @@ export interface ClusterState {
   allowedResources: string[]
 }
 
+/**
+ * Cluster
+ *
+ * @beta
```
Comment (Nokel81) [inline src/main/cluster.ts:55]: This PR should probably have some doc describing what we mean by these markers and what guarantees (or not) they mean.

File: src/common/workspace-store.ts (hunk 2 above)
Comment (jakolehm) [reply src/common/workspace-store.ts:106]: I also think the same but we ran out of time. See #1435.

Comment (Nokel81) [reply src/common/workspace-store.ts:106]: Sounds good

**Review by Nokel81 — APPROVED** [review]

**Review by nevalla — APPROVED** [review]
LGTM

**Pull request merged**